### PR TITLE
Add check dependencies script

### PR DESCRIPTION
**Pull request by YevhenZvieriev**
## Script Overview
Introduces a script for the swift and straightforward installation of Magento 2, specifically tailored for beginners. The script provides helpful recommendations for dependencies.

## Highlighted Features
- Simple interface for selecting the Magento 2 version.
- Dependency recommendations tailored to the chosen Magento version.
- Streamlines the installation process, saving time and effort.

Screenshot_2023-12-07_18-25-55

Screenshot_2023-12-07_18-26-24
## Your contributions, feedback, and suggestions are welcome!

### CODE REVIEW

File: compose/bin/check-dependencies [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+NC='\033[0m' # No Color
+
+declare -A common_dependencies
```
Comment (markshust) [inline compose/bin/check-dependencies:7]: Should this be: `declare -a common_dependencies` ?

File: compose/bin/check-dependencies [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+declare -a magento_versions=("2.4.7-beta2" "2.4.6-p3" "2.4.6-p2" "2.4.6-p1" "2.4.6" "2.4.5-p5" "2.4.5-p4" "2.4.5-p3" "2.4.5-p2" "2.4.5-p1" "2.4.5" "2.4.4-p6" "2.4.4-p5" "2.4.4-p4" "2.4.4-p3" "2.4.4-p2" "2.4.4-p1" "2.4.4" "2.4.3-p3" "2.4.2-p2")
+
+common_dependencies["2.4.7-beta2"]="Composer:2.5 Elasticsearch:8.7 OpenSearch:2.5 MariaDB:10.6 MySQL:8.0 PHP:8.2,8.1 RabbitMQ:3.11 Redis:7.0 Varnish:7.3 Apache:2.4 nginx:1.24 'AWS-Aurora-(MySQL)':8.0 'AWS-S3':✔️ 'AWS-MQ':3.10.20 'AWS-ElastiCache':Redis7.0 'AWS-Elasticsearch':-- 'AWS-OpenSearch':2.5"
```
Comment (markshust) [inline compose/bin/check-dependencies:10]: These lines fail shellcheck validation with:

(( )) doesn't support decimals. Use bc or awk.
See SC2079.

Rule at https://github.com/koalaman/shellcheck/wiki/SC2079

File: compose/bin/check-dependencies [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+print_dependencies() {
+    local version=$1
+    IFS=' ' read -ra dependencies <<< "${common_dependencies[$version]}"
+    for dependency in "${dependencies[@]}"; do
+        echo "  • $dependency"
+    done
+}
+
+read -p "Enter the Magento 2 version you want to install: " magento_version
```
Comment (markshust) [inline compose/bin/check-dependencies:46]: This line fails shellcheck validation with:

read without -r will mangle backslashes.
See SC2162.

Per https://github.com/koalaman/shellcheck/wiki/SC2162

Comment (markshust) [reply compose/bin/check-dependencies:46]: I think we should also rename this from "Enter the Magento 2 version you want to install: " to "Enter the Magento 2 version you'd like to check dependencies on: ", as the former implies that we're about to install Magento, which we are not.

**Review by markshust — CHANGES_REQUESTED** [review]

File: compose/bin/check-dependencies (hunk 3 above)
Comment (YevhenZvieriev) [reply compose/bin/check-dependencies:46]: Thx, I fixed it!

File: compose/bin/check-dependencies (hunk 1 above)
Comment (YevhenZvieriev) [reply compose/bin/check-dependencies:7]: Yes, you're right.

Thx, I fixed it!

File: compose/bin/check-dependencies (hunk 3 above)
Comment (YevhenZvieriev) [reply compose/bin/check-dependencies:46]: > I think we should also rename this from "Enter the Magento 2 version you want to install: " to "Enter the Magento 2 version you'd like to check dependencies on: ", as the former implies that we're about to install Magento, which we are not.

This script was supposed to run before installing Magento, so maybe you misunderstood, this script shows the dependencies, not installing Magento. I changed it.

Comment (markshust) [reply compose/bin/check-dependencies]: The verbiage in this line "Enter the Magento 2 version you want to install:" implies that the user wants to install Magento. I understand this script does not do that, and just checks compatibility. For this reason, we need to update this to "Enter the Magento 2 version you'd like to check dependencies for:"

File: compose/bin/check-dependencies (hunk 1 above)
Comment (markshust) [reply compose/bin/check-dependencies:7]: I didn't get an answer to this code review -- should this be `declare -a common_dependencies` ?

**Review by markshust — CHANGES_REQUESTED** [review]
Few comments that still need answers/modifications for

**Review by markshust — APPROVED** [review]

File: compose/bin/check-dependencies [hunk 4]
```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
```
Comment (markshust) [inline compose/bin/check-dependencies:1]: The shebang should be like all of the other helper scripts: `#!/usr/bin/env bash`

**Review by markshust — CHANGES_REQUESTED** [review]
1. The shebang is incorrect.
2. The related documentation needs to be added to the README and related Makefile.
3. I'm getting errors when running this script:

```
~/Sites/magento246-p4 bin/check-dependencies 
bin/check-dependencies: line 7: declare: -A: invalid option
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]
bin/check-dependencies: line 10: 2.4.7-beta3: syntax error: invalid arithmetic operator (error token is ".4.7-beta3")
bin/check-dependencies: line 11: 2.4.7-beta2: syntax error: invalid arithmetic operator (error token is ".4.7-beta2")
bin/check-dependencies: line 12: 2.4.6-p4: syntax error: invalid arithmetic operator (error token is ".4.6-p4")
bin/check-dependencies: line 13: 2.4.6-p3: syntax error: invalid arithmetic operator (error token is ".4.6-p3")
bin/check-dependencies: line 14: 2.4.6-p2: syntax error: invalid arithmetic operator (error token is ".4.6-p2")
bin/check-dependencies: line 15: 2.4.6-p1: syntax error: invalid arithmetic operator (error token is ".4.6-p1")
bin/check-dependencies: line 16: 2.4.6: syntax error: invalid arithmetic operator (error token is ".4.6")
bin/check-dependencies: line 17: 2.4.5-p5: syntax error: invalid arithmetic operator (error token is ".4.5-p5")
bin/check-dependencies: line 18: 2.4.5-p4: syntax error: invalid arithmetic operator (error token is ".4.5-p4")
bin/check-dependencies: line 19: 2.4.5-p3: syntax error: invalid arithmetic operator (error token is ".4.5-p3")
bin/check-dependencies: line 20: 2.4.5-p2: syntax error: invalid arithmetic operator (error token is ".4.5-p2")
bin/check-dependencies: line 21: 2.4.5-p1: syntax error: invalid arithmetic operator (error token is ".4.5-p1")
bin/check-dependencies: line 22: 2.4.5: syntax error: invalid arithmetic operator (error token is ".4.5")
bin/check-dependencies: line 23: 2.4.4-p6: syntax error: invalid arithmetic operator (error token is ".4.4-p6")
bin/check-dependencies: line 24: 2.4.4-p5: syntax error: invalid arithmetic operator (error token is ".4.4-p5")
bin/check-dependencies: line 25: 2.4.4-p4: syntax error: invalid arithmetic operator (error token is ".4.4-p4")
bin/check-dependencies: line 26: 2.4.4-p3: syntax error: invalid arithmetic operator (error token is ".4.4-p3")
bin/check-dependencies: line 27: 2.4.4-p2: syntax error: invalid arithmetic operator (error token is ".4.4-p2")
bin/check-dependencies: line 28: 2.4.4-p1: syntax error: invalid arithmetic operator (error token is ".4.4-p1")
bin/check-dependencies: line 29: 2.4.4: syntax error: invalid arithmetic operator (error token is ".4.4")
bin/check-dependencies: line 30: 2.4.3-p3: syntax error: invalid arithmetic operator (error token is ".4.3-p3")
bin/check-dependencies: line 31: 2.4.2-p2: syntax error: invalid arithmetic operator (error token is ".4.2-p2")
Enter the Magento 2 version you'd like to check dependencies on:
```

**Comment by YevhenZvieriev** [comment]
1. The shebang was changed from `#!/bin/bash` to `#!/usr/bin/env bash`.

2. I added the documentation to the README and Makefile.

3. Can you check it again?
In my case, the script works perfectly:
```
jenyamba@Acer-Aspire-A515-47:~/Projects/magento247$ bin/check-dependencies 
Enter the Magento 2 version you'd like to check dependencies on:  2.4.6-p4
Recommended software dependencies:
  • Composer:2.2
  • Elasticsearch:8.5,7.17
  • OpenSearch:2.5
  • MariaDB:10.6
  • MySQL:8.0
  • PHP:8.2,8.1
  • RabbitMQ:3.11,3.9
  • Redis:7.0
  • Varnish:7.3
  • Apache:2.4
  • nginx:1.24
  • 'AWS-Aurora-(MySQL)':8.0
  • 'AWS-S3':✔️
  • 'AWS-MQ':3.9.16
  • 'AWS-ElastiCache':Redis6.2
  • 'AWS-Elasticsearch':--
  • 'AWS-OpenSearch':1.2
jenyamba@Acer-Aspire-A515-47:~
```

**Comment by YevhenZvieriev** [comment]
I added compatibility with MacOS in the previous commit, so now it should work on MacOS.

Please feel free to check it.

**Review by markshust — APPROVED** [review]

**Comment by markshust** [comment]
Great, thanks, works as expected! Merging in 👍

**Pull request merged**